### PR TITLE
SioConvPSをPolicyに採用し、実験を実行

### PR DESCRIPTION
**Pull request by Geson-anko**
## 概要

SioConvPSをPolicyに採用したConfigファイルを記述しました。

隠れ状態のdepth数がForward DynamicsとPolicyで異なっても良いようにするために、`StackwiseLinear`を新たに実装して用いました。

## レビューして欲しいファイル

* ami/models/components/stacked_hidden_state.py
* tests/models/components/test_stacked_hidden_state.py
* configs/models/sioconv_policy/default.yaml

## Submit前の確認項目

- [x] タイトルは一目でわかるようにし、説明文は PR を簡潔に説明するようにしましたか?
- [x] あなたの PR が、異なる変更を束ねたものではなく、ひとつのことだけを行うものであることを確認しましたか?
- [x] この PR で導入されたすべての変更点をリストアップしましたか?
- [x] PR を`make run`コマンドでローカルでテストしましたか?

### CODE REVIEW

**Comment by Geson-anko** [comment]
@myxyy 
修正しました。
https://github.com/MLShukai/ami/pull/268/commits/c54dca09e210af4645df0f00a5e7a35e5bd0c7cc
https://github.com/MLShukai/ami/pull/268/commits/606da92ab7dc1a3056ca56455d6e9672ba66258e

**Review by myxyy — APPROVED** [review]

File: ami/models/components/stacked_hidden_state.py [hunk 1]
```diff
@@ -45,3 +45,26 @@ def forward(self, x: Tensor, hidden_stack: Tensor) -> tuple[Tensor, Tensor]:
...
+        self._weight = nn.Parameter(torch.zeros(in_stacks, out_stacks))
+
+    def forward(self, hidden_stack: Tensor) -> Tensor:
+        """
+        Args:
+            hidden_stack (Tensor): shape is (*, in_stacks, dim)
+
+        Returns:
+            Tensor: shape is (*, out_stacks, dim)
+        """
```
Comment (myxyy) [inline ami/models/components/stacked_hidden_state.py:66]: `dim`が既知であれば`hidden_stack.reshape(-1,in_stacks,dim)`してshapeを`bid`にして
`self.hidden_stack_linear = nn.Parameter(torch.randn(in_stack, dim,dim))`を使って
```
hidden_stack_shape = hidden_stack.shape
hidden_stack = torch.einsum("bid,ide->bie", hidden_stack.reshape(-1,in_stacks,dim), self.hidden_stack_linear).reshape(hidden_stack_shape)
```
とかできるとよいです

Comment (Geson-anko) [reply ami/models/components/stacked_hidden_state.py:66]: 今回は StackedHiddenSelectionの後に線形層を挟んでいるので、一旦このまま行きます。
